src/server/utils/response.utils.js

```diff
@@ -1,45 +1,60 @@
-const { resolveFromVersion } = require('./resolve.utils');
+const {resolveFromVersion} = require('./resolve.utils');
 const errors = require('./error.utils');
 
 /**
-* @description When resources are read in the controller functions
-* they all need to respond in a similar manner
-* @function handleSingleReadResponse
-* @param {Express.response} res - Express response object
-* @param {function} next - next function from express middleware
-* @param {string} base - Which spec version is this request coming from
-* @param {T} Resource - Resource class to use for the results
-* @param {object} resource_json - resulting json to be passed in to the class
-*/
-let handleSingleReadResponse = (res, next, base, Resource, resource_json) => {
-	if (resource_json) {
+ * @description When resources are read in the controller functions
+ * they all need to respond in a similar manner
+ * @function handleSingleReadResponse
+ * @param {Express.response} res - Express response object
+ * @param {function} next - next function from express middleware
+ * @param {string} base - Which spec version is this request coming from
+ * @param {T} Resource - Resource class to use for the results
+ * @param {object} resource_json - resulting json to be passed in to the class
+ */
+
+let handleSingleReadResponse = (res, next, base, Resource, resource_json, version_id) => {
+	let {version: resource_version} = resource_json;
+
+	if (resource_json) { //if there's resource found
+		if (version_id !== null) {
+			if ((resource_version) && (resource_version === version_id)) {
```

Comment:
Ok. We don't have to check for null then.

Reply:
I removed the null check, but I just remembered I used the same handleSingleReadResponse to handle both READ and VREAD, meaning we would need to check for null (since READ doesn't send version_id). If not, then would have to create a separate handleSingleVREADResponse.